psydac/linalg/block.py

```diff
@@ -48,6 +49,8 @@ def __init__(self,  *spaces):
         else:
             self._dtype = tuple(s.dtype for s in spaces)
 
+        connectivity       = connectivity or {}
+        self._connectivity = connectivity
```

Comment:
We can use `MappingProxyType` here too:

```suggestion
        self._connectivity          = connectivity or {}
        self._connectivity_readonly = MappingProxyType(self._connectivity)
```

Reply:
done in [7426fa5](https://github.com/pyccel/psydac/pull/213/commits/7426fa50b3353eb00a528005c2fb2be977f6a433)